extension/spark-doris-connector/src/main/java/org/apache/doris/spark/backend/BackendClient.java

```diff
@@ -0,0 +1,218 @@
+            logger.debug("Attempt {} to openScanner {}.", attempt, routing);
+            try {
+                TScanOpenResult result = client.open_scanner(openParams);
+                if (result == null) {
+                    logger.warn("Open scanner result from {} is null.", routing);
+                    continue;
+                }
+                if (!TStatusCode.OK.equals(result.getStatus().getStatus_code())) {
+                    logger.warn("The status of open scanner result from {} is '{}', error message is: {}.",
+                            routing, result.getStatus().getStatus_code(), result.getStatus().getError_msgs());
+                    continue;
+                }
+                return result;
+            } catch (TException e) {
+                logger.warn("Open scanner from {} failed.", routing, e);
+                ex = e;
+            }
+        }
+        logger.error(ErrorMessages.CONNECT_FAILED_MESSAGE, routing);
+        throw new ConnectedFailedException(routing.toString(), ex);
+    }
+
+    /**
+     * get next row batch from Doris BE
+     * @param nextBatchParams thrift struct to required by request
+     * @return scan batch result
+     * @throws ConnectedFailedException throw if cannot connect to Doris BE
+     */
+    public TScanBatchResult getNext(TScanNextBatchParams nextBatchParams) throws ConnectedFailedException {
+        logger.debug("GetNext to '{}', parameter is '{}'.", routing, nextBatchParams);
+        if (!isConnected) {
+            open();
+        }
+        TException ex = null;
+        for (int attempt = 0; attempt < retries; ++attempt) {
+            logger.debug("Attempt {} to getNext {}.", attempt, routing);
+            try {
+                TScanBatchResult result  = client.get_next(nextBatchParams);
+                if (result == null) {
+                    logger.warn("GetNext result from {} is null.", routing);
+                    continue;
+                }
+                if (!TStatusCode.OK.equals(result.getStatus().getStatus_code())) {
+                    logger.warn("The status of get next result from {} is '{}', error message is: {}.",
+                            routing, result.getStatus().getStatus_code(), result.getStatus().getError_msgs());
+                    continue;
+                }
+                return result;
+            } catch (TException e) {
+                logger.warn("Get next from {} failed.", routing, e);
+                ex = e;
+            }
+        }
+        logger.error(ErrorMessages.CONNECT_FAILED_MESSAGE, routing);
+        throw new ConnectedFailedException(routing.toString(), ex);
+    }
+
+    /**
+     * close an scanner.
+     * @param closeParams thrift struct to required by request
+     */
+    public void closeScanner(TScanCloseParams closeParams) {
+        logger.debug("CloseScanner to '{}', parameter is '{}'.", routing, closeParams);
+        if (!isConnected) {
+            try {
+                open();
```

Comment:
Can you explain why the `open()` be invoked here?

Reply:
in case of 'close' be called outside, you never know how user use it